src/daft-connect/src/display.rs

```diff
@@ -0,0 +1,382 @@
+use std::fmt::Write;
+
+use daft_core::prelude::*;
+
+pub trait SparkDisplay {
+    fn repr_spark_string(&self) -> String;
+}
+
+impl SparkDisplay for Schema {
+    fn repr_spark_string(&self) -> String {
+        let mut output = String::new();
+        // Start with root
+        writeln!(&mut output, "root").unwrap();
+
+        // Print each top-level field with indentation level 1
+        for (name, field) in &self.fields {
+            // We'll rely on a helper function that knows how to print a field with given indentation
+            write_field(&mut output, name, &field.dtype, 1).unwrap();
+        }
+        output
+    }
+}
+
+impl SparkDisplay for Field {
+    fn repr_spark_string(&self) -> String {
+        // Fields on their own need context (indentation) to print nicely.
+        // For a standalone Field, we might choose zero indentation or provide a helper method.
+        // Here we choose zero indentation since it's ambiguous outside a schema:
+        let mut output = String::new();
+        write_field(&mut output, &self.name, &self.dtype, 0).unwrap();
+        output
+    }
+}
+
+impl SparkDisplay for DataType {
+    fn repr_spark_string(&self) -> String {
+        type_to_string(self)
+    }
+}
+
+// Private helpers to mimic the original indentation style and recursive printing:
+
+fn write_field(
+    w: &mut String,
+    field_name: &str,
+    dtype: &DataType,
+    level: usize,
+) -> eyre::Result<()> {
+    fn write_field_inner(
+        w: &mut String,
+        field_name: &str,
+        dtype: &DataType,
+        level: usize,
+        is_list: bool,
+    ) -> eyre::Result<()> {
+        /// All daft fields are nullable.
+        const NULLABLE: bool = true;
+
+        let indent = make_indent(level);
+
+        let dtype_str = type_to_string(dtype);
+
+        writeln!(
+            w,
+            "{indent}{field_name}: {dtype_str} ({nullable} = {NULLABLE})",
```

Comment:
If all `daft` fields are always NULLABLE, why don't we just have `... ({nullable} = true)` instead?